plugins/data-management/src/HierarchicalTrackSelectorWidget/model.ts

```diff
@@ -198,35 +298,75 @@ export default function stateTreeFactory(pluginManager: PluginManager) {
           ...filterTracks(getSession(self).tracks, self),
         ].filter(notEmpty)
       },
-    }))
-    .views(self => ({
+
       /**
-       * #method
+       * #getter
+       * filters out tracks that are not in the favorites group
        */
-      connectionHierarchy(connection: {
-        name: string
-        tracks: AnyConfigurationModel[]
-      }): TreeNode[] {
-        return generateHierarchy({
-          model: self,
-          trackConfs: self.connectionTrackConfigurations(connection),
-          extra: connection.name,
-        })
+      get favoriteTracks() {
+        return this.trackConfigurations.filter(track =>
+          self.favorites.includes(readConfObject(track, 'trackId')),
+        )
+      },
+
+      /**
+       * #getter
+       * filters out tracks that are not in the recently used group
+       */
+      get recentlyUsedTracks() {
+        return this.trackConfigurations.filter(track =>
+          self.recentlyUsed.includes(readConfObject(track, 'trackId')),
+        )
       },
     }))
     .views(self => ({
       get allTracks() {
         const { connectionInstances = [] } = getSession(self)
         return [
+          {
+            group: '✨Favorites',
+            tracks: self.favoriteTracks,
+            noCategories: true,
+            menuItems: [
+              {
+                label: 'Clear all favorites',
+                onClick: () => self.clearFavorites(),
+              },
+            ],
+          },
+          {
+            group: '🕒 Recently used',
+            tracks: self.recentlyUsedTracks,
+            isOpenByDefault: false,
+            noCategories: true,
+            menuItems: [
+              {
+                label: 'Clear all recently used',
+                onClick: () => self.clearRecentlyUsed(),
+              },
+            ],
+          },
           {
             group: 'Tracks',
             tracks: self.trackConfigurations,
+            noCategories: false,
+            menuItems: [],
           },
           ...connectionInstances.flatMap(c => ({
             group: getConf(c, 'name'),
             tracks: c.tracks,
+            noCategories: false,
+            menuItems: [],
           })),
-        ]
+        ].filter(
+          // filters out categories favorites and recently used if the user toggles them off
```

Comment:
instead of filtering as a post-process step, could move this up to where they are created and say e.g.

```
...(self.showRecentlyUsedCategory?                  [ {
            group: '✨Favorites',
            tracks: self.favoriteTracks,
            noCategories: true,
            menuItems: [
              {
                label: 'Clear all favorites',
                onClick: () => self.clearFavorites(),
              },
            ],
          }, {
            group: '🕒 Recently used',
            tracks: self.recentlyUsedTracks,
            isOpenByDefault: false,
            noCategories: true,
            menuItems: [
              {
                label: 'Clear all recently used',
                onClick: () => self.clearRecentlyUsed(),
              },
            ],
          }]:[]),
...(self.showFavoritesCategory?          [{
            group: '✨Favorites',
            tracks: self.favoriteTracks,
            noCategories: true,
            menuItems: [
              {
                label: 'Clear all favorites',
                onClick: () => self.clearFavorites(),
              },
            ],
          }]:[])
```